### PR TITLE
br: support to write/ready azure blobs with customer provided key

### PR DESCRIPTION
**Pull request by Leavrth**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
add `encryption-scope` and `encryption-key` for azure blob storage

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/14356
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]

File: br/backup-and-restore-storages.md [hunk 1]
```diff
@@ -80,6 +80,8 @@ BACKUP DATABASE * TO 's3://bucket-name/prefix' SEND_CREDENTIALS_TO_TIKV = FALSE;
     - `account-key`：访问密钥
     - `sas-token`：共享访问签名令牌
     - `access-tier`：上传对象的存储类别，例如 `Hot`、`Cool`、`Archive`，默认值为该存储账户的默认访问层。
+    - `encryption-scope`: 服务端加密，提供加密范围
```
Comment (tonyxuqqi) [inline br/backup-and-restore-storages.md:83]: 范围包含哪些可能的值

Comment (Leavrth) [reply br/backup-and-restore-storages.md:83]: 这个是用户自己在 azure 存储账户里面创建的，名字是用户自定义的

Comment (benmaoer) [reply br/backup-and-restore-storages.md:83]: 链个链接到 Azure 的文档介绍下 encryption-scope 的含义？

Comment (Leavrth) [reply br/backup-and-restore-storages.md]: done

File: encryption-at-rest.md [hunk 2]
```diff
@@ -308,3 +308,73 @@ TiFlash 在 v4.0.9 同样对加密元数据操作进行了优化，其兼容性
...
+```
+./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>?encryption-key=<aes256-key>"
+```
+
+或添加到 BR 的环境变量中
+
+```
+export AZURE_ENCRYPTION_KEY=<aes256-key>
+./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>"
+```
```
Comment (Oreoxmt) [inline encryption-at-rest.md:380]:
````suggestion
## BR Azure Blob Storage 服务端加密

使用 BR 备份数据到 Azure Blob Storage 时，可以选择使用加密范围 (Encryption Scope) 或加密密钥 (Encryption Key) 进行数据的服务端加密。

### 方法一：使用加密范围

你可以通过以下两种方式为备份数据指定加密范围：

- 在 `backup` 命令中添加 `--azblob.encryption-scope` 参数，并设置为加密范围名：

    ```shell
    ./br backup full --pd <pd-address> --storage "azure://<bucket>/<prefix>" --azblob.encryption-scope scope1
    ```

- 在 URI 中添加 `encryption-scope`，并设置为加密范围名：

    ```shell
    ./br backup full --pd <pd-address> --storage "azure://<bucket>/<prefix>?encryption-scope=scope1"
    ```

更多信息请参考 Azure 文档中的[上传具有加密范围的 blob](https://learn.microsoft.com/zh-cn/azure/storage/blobs/encryption-scope-manage?tabs=powershell#upload-a-blob-with-an-encryption-scope)。

在恢复备份时，不需要指定加密范围，Azure Blob Storage 将自动进行解密。示例如下：

```shell
./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>"
```

### 方法二：使用加密密钥

你可以通过以下三种方式为备份数据指定加密密钥：

- 在 `backup` 命令中添加 `--azblob.encryption-key` 参数，并设置为 AES256 加密密钥：

    ```shell
    ./br backup full --pd <pd-address> --storage "azure://<bucket>/<prefix>" --azblob.encryption-key <aes256-key>
    ```

- 在 URI 中添加 `encryption-key`，并设置为 AES256 加密密钥。如果密钥包含 URI 保留字符，例如 `&`、`%` 等，需要先进行百分号编码：

    ```shell
    ./br backup full --pd <pd-address> --storage "azure://<bucket>/<prefix>?encryption-key=<aes256-key>"
    ```

- 在 BR 的环境变量中添加 `AZURE_ENCRYPTION_KEY`，并设置为 AES256 加密密钥。在运行前，请确保环境变量中的加密密钥是已知的，避免忘记密钥。

    ```shell
    export AZURE_ENCRYPTION_KEY=<aes256-key>
    ./br backup full --pd <pd-address> --storage "azure://<bucket>/<prefix>"
    ```

更多信息请参考 Azure 文档中的[在对 Blob 存储的请求中提供加密密钥](https://learn.microsoft.com/zh-cn/azure/storage/blobs/encryption-customer-provided-keys)。

在恢复备份时，需要指定加密密钥。示例如下：

- 在 `restore` 命令中传递 `--azblob.encryption-key` 参数：

    ```shell
    ./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>" --azblob.encryption-key <aes256-key>
    ```

- 在 URI 中添加 `encryption-key`：

    ```shell
    ./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>?encryption-key=<aes256-key>"
    ```

- 在 BR 的环境变量中添加 `AZURE_ENCRYPTION_KEY`：

    ```shell
    export AZURE_ENCRYPTION_KEY=<aes256-key>
    ./br restore full --pd <pd-address> --storage "azure://<bucket>/<prefix>"
    ```
````

**Review by Oreoxmt — COMMENTED** [review]
@Leavrth PTAL

**Review by Oreoxmt — APPROVED** [review]

**Review by hfxsd — APPROVED** [review]

**Pull request merged**